test/python/test_qasm_simulator_projectq.py

```diff
@@ -73,32 +75,45 @@ def setUpClass(cls):
 
     def test_gate_x(self):
         shots = 100
-        qp = QuantumProgram()
-        qr = QuantumRegister(1, "qr")
-        cr = ClassicalRegister(1, "cr")
-        qc = QuantumCircuit(qr, cr)
+        qr = QuantumRegister(1)
+        cr = ClassicalRegister(1)
+        qc = QuantumCircuit(qr, cr, name='test_gate_x')
         qc.x(qr[0])
         qc.measure(qr, cr)
-        qp.add_circuit("circuit_name", qc)
-        result_pq = qp.execute('circuit_name',
-                               backend='local_qasm_simulator_projectq',
-                               seed=1, shots=shots)
-        self.assertEqual(result_pq.get_counts(),
+        cconfig = COMPILE_CONFIG_DEFAULT
+        cconfig['shots'] = shots
+        qobj = qiskit._compiler.compile([qc], pq_simulator,
+                                        compile_config=cconfig)
+        q_job = QuantumJob(qobj, pq_simulator, preformatted=True,
+                           resources={'max_credits': qobj['config']['max_credits'],
+                                      'wait': 5,
+                                      'timeout': 30})
+        job = pq_simulator.run(q_job)
+        result_pq = job.result(timeout=30)
```

Comment:
What is the relation between `timeout` as an argument of `job.result()`, and `timeout` as one of the `resources` of a QuantumJob? Should the latter one go away?

Reply:
In particular, the `timeout` in QuantumJob seems to really cause timeouts in the job, which is not expected from an async job. When I set `timeout` in QuantumJob to a very low number, then `job.status` shows `DONE: 'job has successfully run'` after a short time, but `job.result()._result` shows `QISkit Time Out'`

Reply:
hmm, I'll see if I can clean this up. As an aside I noticed that when passing a qobj to the init of QuantumJob it seems like the other keyword parameters (e.g. timeout, shots) are ignored.

Reply:
So I guess you removed the QuantumJob `timeout` in [84d6348](https://github.com/QISKit/qiskit-sdk-py/pull/403/commits/84d6348f579eeab3a4c566e55b3b23831a94e631)?
In that case can you remove its remaining uses in `test_qasm_simulator_projectq`, `test_qasm_simulator_py`, `test_statevector_simulator_sympy`?